source/dlangbot/git.d

```diff
@@ -0,0 +1,55 @@
+module dlangbot.git;
+
+import std.conv, std.file, std.path, std.string, std.uuid;
+import std.format, std.stdio;
+
+import dlangbot.github;
+import vibe.core.log;
+
+string gitURL = "http://0.0.0.0:9006";
+
+void rebase(PullRequest* pr)
+{
+    import std.process;
+    auto uniqDir = tempDir.buildPath("dlang-bot-git", randomUUID.to!string.replace("-", ""));
+    uniqDir.mkdirRecurse;
+    scope(exit) uniqDir.rmdirRecurse;
+
+    auto targetBranch = pr.base.ref_;
+    auto remoteDir = pr.repoURL;
+
+    logInfo("[git/%s]: cloning branch %s...", pr.repoSlug, targetBranch);
+    auto pid = spawnShell("git clone -b %s %s %s".format(targetBranch, remoteDir, uniqDir));
```

Comment:
spawnProcess > spawnShell in all situations, unless you need to use a shell built-in or run a user-supplied shell command